test/functional/p2p_ibd_stalling.py

```diff
@@ -135,21 +134,16 @@ def run_test(self):
         self.wait_until(lambda: self.is_block_requested(peers, stall_block))
         self.all_sync_send_with_ping(peers)
 
-        self.log.info("Provide the withheld block and check that stalling timeout gets reduced back to 2 seconds")
-        with node.assert_debug_log(expected_msgs=['Decreased stalling timeout to 2 seconds']):
+        self.log.info("Provide the first withheld block and check that stalling timeout gets reduced back to 2 seconds")
+        with node.assert_debug_log(expected_msgs=['Decreased stalling timeout to 2 seconds'], unexpected_msgs=['Stall started']):
             for p in peers:
                 if p.is_connected and (stall_block in p.getdata_requests):
                     p.send_without_ping(msg_block(block_dict[stall_block]))
+            self.all_sync_send_with_ping(peers)
 
-        self.log.info("Check that all outstanding blocks get connected")
-        self.wait_until(lambda: node.getblockcount() == NUM_BLOCKS)
+        self.log.info("Check that all outstanding blocks up to the second stall block get connected")
+        self.wait_until(lambda: node.getblockcount() == 500)
```

Comment:
nit: could use a variable `second_stall_block_index = 500` or something like that to avoid repeating the number `500`.

Reply:
changed to use one var for each index, plus one array.